src/toxcore/tcp/server/server.rs

```diff
@@ -99,13 +100,20 @@ impl Server {
     */
     pub fn shutdown_client(&self, pk: &PublicKey) -> IoFuture<()> {
         let mut state = self.state.write();
+        let state = state.deref_mut();
```

Comment:
Do we really need to use deref_mut op here?

Reply:
It just used for making type of state to simple.

Reply:
Ok I will remove it.

Reply:
done